src/sentry/api/serializers/models/group.py

```diff
@@ -531,16 +531,25 @@ def query_tsdb(self, group_ids, query_params):
 
     def get_stats(self, item_list, user):
         if self.stats_period:
-            # we need to compute stats at 1d (1h resolution), and 14d
+            # we need to compute stats at 1d (1h resolution), and 14d or a custom given period
             group_ids = [g.id for g in item_list]
 
-            segments, interval = self.STATS_PERIOD_CHOICES[self.stats_period]
-            now = timezone.now()
-            query_params = {
-                "start": now - ((segments - 1) * interval),
-                "end": now,
-                "rollup": int(interval.total_seconds()),
-            }
+            if self.stats_period == "auto":
+                query_params = {
+                    "start": self.stats_period_start,
+                    "end": self.stats_period_end,
+                    "rollup": int(
+                        (self.stats_period_end - self.stats_period_start).total_seconds() / 29
```

Comment:
Why 29?

Reply:
So that tsdb returns us stats divided into 30 segments

Reply:
But I am actually trying to improve this logic here

Reply:
Giving the 29 a name could help with that.

Reply:
Yeah that makes sense